### DISCUSSION TITLE
[LGTM] C/C++ extraction always uses Python 2

### DISCUSSION THREAD

**Discussion by QuLogic** [github/securitylab#383] [Q&A] [answered]
When performing C/C++ extraction, LGTM tries a bunch of things, and apparently does `python setup.py build`. This is broken on packages that do not support Python 2, e.g., https://lgtm.com/projects/g/matplotlib/matplotlib/logs/languages/lang:cpp

I tried [explicitly setting Python 3](https://github.com/matplotlib/matplotlib/pull/20454), but this apparently did not affect the C/C++ side.

PS, why is it so hard to figure out where to report issues with LGTM? I'm not even sure if this is the right place.

**Comment by tausbn** [comment, marked as answer]
For this particular case, you could try one of two things

1. Invoking `python3 setup.py build` directly by setting the [`configure`](https://lgtm.com/help/lgtm/cpp-extraction#cpp-configure-step) step for `cpp`
2. Explicitly modifying the `PATH` variable to put the Python 3 interpreter first. For this, something like putting
```
cp `which python3` $LGTM_WORKSPACE/python
PATH=$LGTM_WORKSPACE:$PATH
```
in the `before_index` step might work.

(Note: I have not tested the two approaches above. If they _don't_ work, do let us know.)

**Reply by QuLogic** [reply]
I tried a [test build](https://lgtm.com/logs/d97d57563c2959563937736e19dd93194fd5291e/lang:cpp) with the first option, which seems to work. Now I need to figure out how to get the download part working.
